Add image optimization

maier49 opened the pull request:
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/)
* [ ] Unit or Functional tests are included in the PR
* [x] schema.json has been updated appopriately

<!--
Our bots should ensure:

* [ ] All contributors have signed a CLA
* [ ] The PR passes CI testing
* [ ] Code coverage is maintained
* [ ] The PR has been reviewed and approved
-->

**Description:**
Proposed solution for adding the image-webpack-loader and related config to the build.
* Only optimizes images if the `imageOptimization` flag is present and for dist builds. This can be an expensive process so opting in seems reasonable.
* The image-webpack-loader defers to multiple different tools for optimizing different types of images, and they take a wide variety of arguments. Rather than try to abstract over all those different APIs this just accepts an object for each tool and passes those options directly.
Resolves #328

matt-gadd commented:
@maier49 if we are going to allow a direct pass through can you please document the options and improve the schema, we shouldn't have to forward users to a webpack loader page for this. I'm going to presume this at least does optimization with some defaults being config free? Seems a bit weird to turn it on you have to specify an empty object rather than a boolean from what I can see in the schema?

maier49 commented:
@matt-gadd I've updated the schema to allow you to simply pass `true` to enable image optimization with default settings, and yes we are doing optimization with default config free.

I also added a section to the documentation for this. I added the `enabled` flag that can turn off (or on for webp) each specific optimization library to the schema, but if you take a look at the options APIs for each of those libraries adding all of that to our schema seems excessive to me. The loader itself takes the same approach as I have now in the docs and links out to each of the libraries individually. If you still want me to though I can add in the options for each.
